Documentation/network/servicemesh/mTLS-auth/mTLS-auth.rst

```diff
@@ -0,0 +1,101 @@
+MTLS auth therefore requires a framework secure identity verification for distributed systems.
+
+In Cilium's current mTLS auth support, that is provided through SPIFFE (Secure Production Identity Framework for Everyone).
+
+Here are some of the benefits provided by SPIFFE:
+
+- Trustworthy identity issuance: SPIFFE provides a standardized mechanism for issuing and managing identities. 
+  It ensures that each service in a distributed system receives a unique and verifiable identity, even in dynamic environments where services may scale up or down frequently.
+- Identity federation: SPIFFE enables identity federation, allowing services from different organizations or 
+  administrative domains to authenticate and establish trust. This is particularly valuable in scenarios where multiple organizations collaborate or interact with each other's services.
+- Identity attestation: SPIFFE allows services to prove their identities through attestation. 
+  It ensures that services can demonstrate their authenticity and integrity by providing verifiable evidence about their identity, such as digital signatures or cryptographic proofs.
+- Dynamic and scalable environments: SPIFFE addresses the challenges of identity management in dynamic environments. 
+  It supports automatic identity issuance, rotation, and revocation, which are critical in cloud-native architectures where services may be constantly deployed, updated, or retired.
+
+By combining mTLS with SPIFFE, we establish a robust and scalable security infrastructure that provides strong mutual authentication and verifiable identities in dynamic distributed systems.
+
+Cilium and SPIFFE
+#################
+
+SPIFFE provides an API model that allows workloads to request an identity from a central server. In our case, a workload means the same thing that a Cilium Security Identity does - a set of pods described by a label set. 
+A SPIFFE identity is a subclass of URI, and looks something like this: ``spiffe://trust.domain/path/with/encoded/info``.
+
+There are two main parts of a SPIFFE setup:
+
+- A central SPIRE server, which forms the root of trust for the trust domain.
+- A per-node SPIRE agent, which first gets its own identity from the SPIRE server, then validates the identity requests of workloads running on its node.
+
+When a workload wants to get its identity, usually at startup, it connects to the local SPIRE agent using the SPIFFE workload API, and describes itself to the agent.
+
+The SPIRE agent then checks that the workload is really who it says it is, and then connects to the SPIRE server and attests that the workload is requesting an identity, and that the request is valid. 
+
+The SPIRE agent checks a number of things about the workload, that the pod is actually running on the node it’s coming from, that the labels match, and so on. 
```

Comment:
can we be explicit about what parameters are checked? it maybe helpful for troubleshooting purposes